src/synthesizer/load_data/load_eagle.py

```diff
@@ -0,0 +1,932 @@
+        if dat.ndim == 0:
+            return np.array([])
+
+    return dat
+
+
+def read_array(
+    ftype: str,
+    directory: str,
+    tag: str,
+    dataset: str,
+    numThreads: int = 1,
+    noH: bool = False,
+    physicalUnits: bool = False,
+    CGS: bool = False,
+    photUnits: bool = False,
+    verbose: bool = True,
+) -> Union[NDArray[Any], unyt_array[unyt_quantity]]:
+    """
+
+    Arguments:
+        ftype (str)
+            eagle file type to read
+        directory (str)
+            location of the eagle simulation directory
+        tag (str)
+            snapshot tag to read
+        dataset (str)
+            name of the dataset to read
+        numThreads (int)
+            number threads to use
+        noH (bool)
+            remove any reduced Hubble factors
+        physicalUnits (bool)
+            return in physical units
+        CGS (bool)
+            return in CGS units
+        verbose (bool)
+            verbose condition
+    """
+    start = timeit.default_timer()
+
+    files = get_files(ftype, directory, tag)
+
+    if numThreads == 1:
+        pool = schwimmbad.SerialPool()
+    elif numThreads == -1:
+        pool = schwimmbad.MultiPool()
+    else:
+        pool = schwimmbad.MultiPool(processes=numThreads)
+
+    lg = partial(read_hdf5, dataset=dataset)
+    dat = list(pool.map(lg, files))
+
+    # get indices of non-empty arrays
+    non_empty_file_indices = [i for i, d in enumerate(dat) if d.shape[0] != 0]
+
+    # ignore files with no data
+    dat = [d for d in dat if d.shape[0] != 0]
+    dat = np.concatenate(dat, axis=0)
+
+    pool.close()
+
+    stop = timeit.default_timer()
+
+    if verbose:
+        print(
+            "Reading in '{}' for z = {} using {} thread(s) took {}s".format(
+                dataset,
+                np.round(
+                    read_header(ftype, directory, tag, dataset="Redshift"), 3
+                ),
+                numThreads,
+                np.round(stop - start, 6),
+            )
+        )
+
+    if noH:
+        dat = apply_hfreeUnits_conversion(
+            files[non_empty_file_indices[0]], dataset, dat, verbose=verbose
+        )
+
+    if physicalUnits:
+        dat = apply_physicalUnits_conversion(
+            files[non_empty_file_indices[0]], dataset, dat, verbose=verbose
+        )
+    if CGS:
+        dat = apply_CGSUnits_conversion(
+            files[non_empty_file_indices[0]], dataset, dat, verbose=verbose
+        )
+    if photUnits:
+        dat = apply_PhotUnits(
+            files[non_empty_file_indices[0]], dataset, dat, verbose=verbose
+        )
+    return dat
+
+
+def read_header(
+    ftype: str,
+    directory: str,
+    tag: str,
+    dataset: str,
+) -> NDArray[Any]:
+    """
+    ftype (str)
+        eagle file type to read
+    directory (str)
+        location of the eagle simulation directory
+    tag (str)
+        snapshot tag to read
+    dataset (str)
+        name of the dataset to read
```

Comment:
Missing Return (but I'm not going to point anymore out, the doc strings need the Returns: block)